glib/gobject-sys/build_manual.rs

```diff
@@ -0,0 +1,33 @@
+#[cfg(not(docsrs))]
+use std::{env, fs::File, io::Write, path::Path};
+
+#[cfg(not(docsrs))]
+pub fn main() {
+    let deps = system_deps::Config::new()
+        .probe()
+        .expect("if this module is called, this is valid");
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let out_path = Path::new(&out_dir).join("param_spec.rs");
+
+    // Generating a link attribute is necessary in windows
```

Comment:
Did you check if e.g. gtk4-rs or gstreamer-rs are also accessing some library constant? That would have the same issue

Reply:
Yes, I tested gstreamer-rs and I just checked gtk4-rs, and the only use seems to be on glib. At least on github [no other crates seem to use this specific constant](https://github.com/search?q=g_param_spec_types+language%3Arust&type=code).

Reply:
They do use constants, but they are generated by gir and are all defined as `pub const ...`. The thing with this one is that it is not added to gir files and that it is defined but not initialized in the header https://gitlab.gnome.org/GNOME/glib/-/blob/main/gobject/gparamspecs.h#L1147. So maybe that has to do with it?

Reply:
Ok, I think I found the difference between the `#[link]` attribute and adding them on the command line. Here is the code that generates libraries with the attribute (https://github.com/rust-lang/rust/blob/170d6cb845c8c3f0dcec5cdd4210df9ecf990244/compiler/rustc_metadata/src/native_libs.rs#L452), which sets `foreign_modules` and `dll_imports`. If we rename the libraries when they have the link attribute already set, they use the existing metadata and it works. However, when adding a library that is not mentioned in `#[link]`, the dll imports is not set (https://github.com/rust-lang/rust/blob/170d6cb845c8c3f0dcec5cdd4210df9ecf990244/compiler/rustc_metadata/src/native_libs.rs#L541).

I'm not sure how feasible it is to modify the behaviour of the linking there, because it needs access to the foreign items in the module to call dllimport, and it is doing that with the macro.